compiler/main.bal

```diff
@@ -39,17 +39,29 @@ public function main(string[] filenames, *Options opts) returns error? {
         if filenames.length() > 1 {
             return error("multiple input files not supported with --showTypes");
         }
-        check printDiagnostic(showTypes([{ filename: filenames[0] }]));
-        return;
+        CompileError? err = showTypes([{ filename: filenames[0] }]);
+        if err is err:Diagnostic {
+            printDiagnostic(err);
+            return error(""); // we don't need to show any thing for this error because printDiagnostic already does it;
+        }
+        return err;
     }
     nback:Options nbackOptions = {
         gcName: check nback:validGcName(opts.gc),
         debugLevel: check nback:validDebugLevel(opts.debugLevel)
     };
+    int errorFileCount = 0;
     foreach string filename in filenames {
         var [basename, ext] = basenameExtension(filename);
         if ext == SOURCE_EXTENSION {
-            check printDiagnostic(compileBalFile(filename, basename, check chooseOutputBasename(basename, opts.outDir), nbackOptions, opts));
+            CompileError? err = compileBalFile(filename, basename, check chooseOutputBasename(basename, opts.outDir), nbackOptions, opts);
+            if err is err:Diagnostic {
+                printDiagnostic(err);
+                errorFileCount += 1;
+            }
+            else if err != () {
```

Comment:
Please do the change I suggested in the previous review.

Reply:
Currently throw a compiler error. Created [issue](34014)